tutorial/consuming_packages/getting_started/the_flexibility_of_conanfile_py.rst

```diff
@@ -0,0 +1,337 @@
+  **CMakeDeps** as in the *conanfile.txt*.
+* **requirements()** in this method we can use the ``self.requires()`` and
+  ``self.tool_requires()`` methods to declare all our dependencies (libraries and build
+  tools).
+
+You can check that running the same commands as in the previous examples will lead to the
+same results as before.
+
+.. code-block:: bash
+    :caption: Windows
+
+    $ conan install . --output-folder=build --build=missing
+    $ cd build
+    $ conanbuild.bat
+    # assuming Visual Studio 15 2017 is your VS version and that it matches your default profile
+    $ cmake .. -G "Visual Studio 15 2017" -DCMAKE_TOOLCHAIN_FILE=conan_toolchain.cmake
+    $ cmake --build . --config Release
+    ...
+    Building with CMake version: 3.19.8
+    ...
+    [100%] Built target compressor
+
+    $ Release\compressor.exe
+    Uncompressed size is: 233
+    Compressed size is: 147
+    ZLIB VERSION: 1.2.11
+    $ deactivate_conanbuild.bat
+
+.. code-block:: bash
+    :caption: Linux, macOS
+    
+    $ conan install . --output-folder cmake-build-release --build=missing
+    $ cd cmake-build-release
+    $ source conanbuild.sh
+    Capturing current environment in deactivate_conanbuildenv-release-x86_64.sh
+    Configuring environment variables    
+    $ cmake .. -DCMAKE_TOOLCHAIN_FILE=conan_toolchain.cmake
+    $ cmake --build .
+    ...
+    Building with CMake version: 3.19.8
+    ...
+    [100%] Built target compressor
+
+    $ ./compressor
+    Uncompressed size is: 233
+    Compressed size is: 147
+    ZLIB VERSION: 1.2.11
+    $ source deactivate_conanbuild.sh
+
+So far we have achieved the same functionality we had using a *conanfile.txt*, let's see
+how we can take advantage of the capabilities of the *conanfile.py* to define the project
+structure we want to follow and also to add some logic using Conan settings and options.
+
+Use the layout() method
+-----------------------
+
+In the previous examples, everytime we executed a `conan install` command we had to use
+the `--output-folder argument` to define where we wanted to create the files that Conan
+generates. Also, note that we used a different folder when building in Windows or in
+Linux/Macos depending if we were using a multi-config CMake generator or not. You can
+define this directly in the `conanfile.py` inside the `layout()` method and make it work
+for every platform without adding more changes:
+
+.. code-block:: python
+    :caption: **conanfile.py**
+
+    from conan import ConanFile
+
+
+    class CompressorRecipe(ConanFile):
+        settings = "os", "compiler", "build_type", "arch"
+        generators = "CMakeToolchain", "CMakeDeps"
+
+        def requirements(self):
+            self.requires("zlib/1.2.11")
+            self.tool_requires("cmake/3.19.8")
+
+        def layout(self):
+            build_type = str(self.settings.build_type)
+            compiler = self.settings.get_safe("compiler")
+            
+            if compiler == "msvc":
```

Comment:
This is a bit misleading, because the layout doesn't depend on the compiler, but the build system. Maybe put a comment clarifying this, and pointing to cmake_layout() and others